frontend/src/components/admin-page/edit-user.tsx

```diff
@@ -0,0 +1,159 @@
+import * as React from 'react';
+import { useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { ButtonColor, ButtonStyleType, ButtonType, EditUserPayloadKey, InputColor, InputType, UserSex, UserType } from 'common/enums';
+import styles from './styles.module.scss';
+import { IEditUserPayload, IUser } from 'common/interfaces';
+import { validationEditUser } from 'validation-schemas';
+import { Button, DateInput, Select, TextInput } from 'components/common';
+import { createOptions } from 'helpers';
+
+interface IProps {
+  user: IUser;
+  func: (userData: IEditUserPayload) => void;
+  hideForm: () => void;
+}
+
+const genderOptions = createOptions<string>(Object.values(UserSex))
+const userTypeOptions = createOptions<string>(Object.values(UserType))
+
+const EditUser: React.FC<IProps> = ({ user, func, hideForm }) => {
+
+  const userWithDate = {...user,birthdate:new Date(user.birthdate)};
```

Comment:
Something wrong. You don't need do it

Reply:
If i don`t do this then react-datepicker returns an error due to an invalid value

Reply:
Just map users from the server